File: NhibernateEntity/FRegions.vb
Imports System.Collections

Namespace NhibernateEntity
#Region "FRegion"

    ''' <summary>
    ''' FRegion object for NHibernate mapped table 'FRegions'.
    ''' </summary>
    Public Class FRegion
#Region "Member Variables"

        Protected _id As String
        Protected _names As String
        Protected _fartherID As String
        Protected _depth As Integer
        Protected _path As String
        Protected _isLeaf As Integer
        Protected _islock As Integer
        Protected _sort As Integer

#End Region

#Region "Constructors"

        Public Sub New()
        End Sub

        Public Sub New(ByVal names As String, ByVal fartherID As String, ByVal depth As Integer, ByVal path As String, ByVal isLeaf As Integer, ByVal islock As Integer, _
         ByVal sort As Integer)
            Me._names = names
            Me._fartherID = fartherID
            Me._depth = depth
            Me._path = path
            Me._isLeaf = isLeaf
            Me._islock = islock
            Me._sort = sort
        End Sub

#End Region

#Region "Public Properties"

        Public Overridable Property Id() As String
            Get
                Return _id
            End Get
            Set(ByVal value As String)
                If value IsNot Nothing AndAlso value.Length > 10 Then
                    Throw New ArgumentOutOfRangeException("Invalid value for Id", value, value.ToString())
                End If
                _id = value
            End Set
        End Property

        Public Overridable Property Names() As String
            Get
                Return _names
            End Get
            Set(ByVal value As String)
                If value IsNot Nothing AndAlso value.Length > 50 Then
                    Throw New ArgumentOutOfRangeException("Invalid value for Names", value, value.ToString())
                End If
                _names = value
            End Set
        End Property

        Public Overridable Property FartherID() As String
            Get
                Return _fartherID
            End Get
            Set(ByVal value As String)
                If value IsNot Nothing AndAlso value.Length > 10 Then
                    Throw New ArgumentOutOfRangeException("Invalid value for FartherID", value, value.ToString())
                End If
                _fartherID = value
            End Set
        End Property

        Public Overridable Property Depth() As Integer
            Get
                Return _depth
            End Get
            Set(ByVal value As Integer)
                _depth = value
            End Set
        End Property

        Public Overridable Property Path() As String
            Get
                Return _path
            End Get
            Set(ByVal value As String)
                If value IsNot Nothing AndAlso value.Length > 150 Then
                    Throw New ArgumentOutOfRangeException("Invalid value for Path", value, value.ToString())
                End If
                _path = value
            End Set
        End Property

        Public Overridable Property IsLeaf() As Integer
            Get
                Return _isLeaf
            End Get
            Set(ByVal value As Integer)
                _isLeaf = value
            End Set
        End Property

        Public Overridable Property Islock() As Integer
            Get
                Return _islock
            End Get
            Set(ByVal value As Integer)
                _islock = value
            End Set
        End Property

        Public Overridable Property Sort() As Integer
            Get
                Return _sort
            End Get
            Set(ByVal value As Integer)
                _sort = value
            End Set
        End Property



#End Region
    End Class
#End Region
End Namespace

File: NhibernateEntity/FRegions.cs
using System;
using System.Collections;

namespace NhibernateEntity
{
	#region FRegion

	/// <summary>
	/// FRegion object for NHibernate mapped table 'FRegions'.
	/// </summary>
	public class FRegion
	{
		#region Member Variables
		
		protected string _id;
		protected string _names;
		protected string _fartherID;
		protected int _depth;
		protected string _path;
		protected int _isLeaf;
		protected int _islock;
		protected int _sort;

		#endregion

		#region Constructors

		public FRegion() { }

		public FRegion( string names, string fartherID, int depth, string path, int isLeaf, int islock, int sort )
		{
			this._names = names;
			this._fartherID = fartherID;
			this._depth = depth;
			this._path = path;
			this._isLeaf = isLeaf;
			this._islock = islock;
			this._sort = sort;
		}

		#endregion

		#region Public Properties

		public string Id
		{
			get {return _id;}
			set
			{
				if ( value != null && value.Length > 10)
					throw new ArgumentOutOfRangeException("Invalid value for Id", value, value.ToString());
				_id = value;
			}
		}

		public string Names
		{
			get { return _names; }
			set
			{
				if ( value != null && value.Length > 50)
					throw new ArgumentOutOfRangeException("Invalid value for Names", value, value.ToString());
				_names = value;
			}
		}

		public string FartherID
		{
			get { return _fartherID; }
			set
			{
				if ( value != null && value.Length > 10)
					throw new ArgumentOutOfRangeException("Invalid value for FartherID", value, value.ToString());
				_fartherID = value;
			}
		}

		public int Depth
		{
			get { return _depth; }
			set { _depth = value; }
		}

		public string Path
		{
			get { return _path; }
			set
			{
				if ( value != null && value.Length > 150)
					throw new ArgumentOutOfRangeException("Invalid value for Path", value, value.ToString());
				_path = value;
			}
		}

		public int IsLeaf
		{
			get { return _isLeaf; }
			set { _isLeaf = value; }
		}

		public int Islock
		{
			get { return _islock; }
			set { _islock = value; }
		}

		public int Sort
		{
			get { return _sort; }
			set { _sort = value; }
		}

		

		#endregion
	}
	#endregion
}